syntax = "proto3";
package mars.stn;


option java_package = "win.liyufan.im.proto";
option java_outer_classname = "GroupOuterClass";

//option objc_class_prefix = "LYF";

enum GroupType {
  //member can add quit change group name and portrait, owner can do all the operations
  GroupType_Normal = 0;
  //every member can add quit change group name and portrait, no one can kickoff others
  GroupType_Free = 1;
  //member can only quit, owner can do all the operations
  GroupType_Restricted = 2;
}

message GroupInfo {
  string target_id = 1;
  string name = 2;
  string portrait = 3;
  string owner = 4;
  GroupType type = 5;
  string extra = 6;
  int64 update_dt = 7;
}

message GroupMember {
  string member_id = 1;
  string alias = 2;
  //0, normal member; 1, manager; 2, owner
  int32 type = 3;	
  int64 update_dt = 4;
}

message Group {
  GroupInfo group_info = 1;
  repeated GroupMember members = 2;
}
